src/super_gradients/training/utils/detection_utils.py

```diff
@@ -664,30 +663,42 @@ class DetectionCollateFN:
     """
 
     def __call__(self, data) -> Tuple[torch.Tensor, torch.Tensor]:
-        batch = default_collate(data)
-        ims, targets = batch[0:2]
-        return ims, self._format_targets(targets)
+        images_batch, labels_batch = list(zip(*data))
+        return self._format_images(images_batch), self._format_targets(labels_batch)
 
-    def _format_targets(self, targets: torch.Tensor) -> torch.Tensor:
-        nlabel = (targets.sum(dim=2) > 0).sum(dim=1)  # number of label per image
-        targets_merged = []
-        for i in range(targets.shape[0]):
-            targets_im = targets[i, : nlabel[i]]
-            batch_column = targets.new_ones((targets_im.shape[0], 1)) * i
-            targets_merged.append(torch.cat((batch_column, targets_im), 1))
-        return torch.cat(targets_merged, 0)
+    def _format_images(self, images_batch: List[Union[torch.Tensor, np.array]]) -> torch.Tensor:
+        images_batch = [torch.tensor(img) for img in images_batch]
+        images_batch_stack = torch.stack(images_batch, 0)
+        if images_batch_stack.shape[3] == 3:
```

Comment:
Let's not introduce HWC->CHW here, since it's quite unexpected responsibility of collate fn. We have permutation transform that does this in an explicit manner.

Reply:
@BloodAxe could you please follow-up on this?